pkg/rp/util/authclient/awsirsa.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authclient
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/google/uuid"
+	ucp_aws "github.com/radius-project/radius/pkg/ucp/aws"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+var _ AuthClient = (*awsIRSA)(nil)
+
+type awsIRSA struct {
+	roleARN string
+}
+
+func NewAwsIRSA(roleARN string) AuthClient {
+	return &awsIRSA{roleARN: roleARN}
+}
+
+func (b *awsIRSA) GetAuthClient(ctx context.Context, templatePath string) (remote.Client, error) {
```

Comment:
Pls add comment for IRSA implementation. it was added for Azure WI, BasicAuth

Reply:
updated it